src/test/scala/is/hail/expr/ir/AggregatorsSuite.scala

```diff
@@ -150,6 +150,17 @@ class AggregatorsSuite {
       args = FastIndexedSeq(I32(2)))
   }
 
+  @Test def takeBy() {
+    runAggregator(TakeBy(), TStruct("x" -> TInt32(), "y" -> TInt32()),
+      FastIndexedSeq(Row(3, 4), Row(null, null),
+      Row(null, 3), Row(21, 2), Row(11, 0), Row(45, 1)),
+      FastIndexedSeq(11, 45),
```

Comment:
More testing ideas: value null, also fewer elements that we're taking (as well as more).